numismatic/feeds/poloniex.py

```diff
@@ -59,68 +92,61 @@ def _handle_trade(msg, subscription):
             which is a Trade entry (t) and is defined as 
             [trade, tradeId, 0/1 (sell/buy), price, amount, timestamp]
         '''
-        if len(msg) != 3:
-            return
-        if len(msg[2]) != 2:
-            return
-
-        # Only interested in trade messages
-        if msg[2][1][0] != 't':
-           return
-        trade_msg = msg[2][1]
-        
-        event_msg = Trade(exchange=subscription.exchange,
-                    symbol=subscription.symbol, 
-                    price=float(trade_msg[3]),
-                    volume=float(trade_msg[4]),
-                    type='SELL' if trade_msg[2] == 0 else 'BUY',
-                    timestamp=trade_msg[5],
-                    id=trade_msg[1],) #  id comes from the original message
-
-
-        subscription.event_stream.emit(event_msg)
+
+        if msg[2][1][0] == 't':
```

Comment:
Like above, include the other checks above otherwise this will raise an Exception if the format of a message is different, for example for a Heartbeat message or something like that.

Reply:
You have a good point. But once I iterate over the messages, this should not be such a big problem.